models/perm/access/repo_permission.go

```diff
@@ -24,6 +24,8 @@ type Permission struct {
 
 	units     []*repo_model.RepoUnit
 	unitsMode map[unit.Type]perm_model.AccessMode
+
+	everyoneAccessMode map[unit.Type]perm_model.AccessMode
```

Comment:
Why not use `unitsMode` directly? If one user is a team member or owner or collaboration, `unitsMode` will read from database. If it's others which matched `everyone`, `unitsMode` could load the data from `repo_unit`.

Reply:
To make the HasAccess has the old behavior